dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/webapp/WebApp.java

```diff
@@ -1052,18 +1052,29 @@ protected void addClassToHandlesTypesStartupSet(String targetClassName, Set<Clas
       
       boolean enableTrace = ( com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable (Level.FINE) );
 
+      int servletSpecLevel = com.ibm.ws.webcontainer.osgi.WebContainer.getServletContainerSpecLevel();
       //check if excluded because the subclasses includes this... and could be more than we want
       //might need to exclude something else like javax.* or java*
-      if (targetClassName.startsWith("java")) {
-          if ((targetClassName.charAt(4) == '.') || (targetClassName.startsWith("javax."))) {
-              if ( enableTrace ) { 
-                  logger.logp(Level.FINE, CLASS_NAME, methodName,
-                              "Internal class, {0}, is not added to the ServletContainerInitializers in the application: {1}",
-                              new Object[] { targetClassName, this.config.getDisplayName() });
-              }
-              return;
+      if(servletSpecLevel <= 40){
```

Comment:
Can you use the constant instead: com.ibm.ws.webcontainer.osgi.WebContainer.SPEC_LEVEL_40

Reply:
Changed